aten/src/ATen/native/mkldnn/xpu/detail/QMatmul.cpp

```diff
@@ -0,0 +1,281 @@
+#include <ATen/Tensor.h>
+#include <ATen/core/Tensor.h>
+#include <c10/core/ScalarType.h>
+
+#include <ATen/native/mkldnn/xpu/detail/Attr.h>
+#include <ATen/native/mkldnn/xpu/detail/oneDNNContext.h>
+
+#include <oneapi/dnnl/dnnl.hpp>
+
+namespace at::native::onednn {
+
+void quantized_matmul(
+    at::Tensor mat1, // act
+    double input_scale,
+    int64_t input_zero_point,
+    at::Tensor mat2, // weight
+    at::Tensor& weight_scales,
+    at::Tensor& weight_zero_points,
+    at::Tensor& b_raw,
+    at::Tensor result, // output
+    double output_scale,
+    int64_t output_zero_point,
+    std::optional<c10::ScalarType> output_dtype,
+    std::optional<at::Tensor> other, // extra input for binary-post-op
+    double other_scale,
+    int64_t other_zero_point,
+    const c10::string_view& binary_post_op,
+    double binary_alpha,
+    const c10::string_view& unary_post_op,
+    torch::List<std::optional<at::Scalar>>& unary_post_op_args,
+    c10::string_view unary_post_op_algorithm) {
+  // [Note] Quantized Matrix Multiplication at XPU
+  // The following code integrates oneDNN quantized gemm. The quantization
+  // config we support:
+  // activation: s8&u8; per tensor calibrated; symmetric&asymmetric
+  // weight: s8; per_tensor/per_channel calibrated; symmetric
+  bool m2_trans = true;
```

Comment:
It is always `True` as no code sets it as `False`.

Reply:
Here is due to we do de weight transpose in `qlinear_prepack`. So the weight should be always use transposed stride. I will move m2_trans to to function argument list, as it is better to let users of this function to decide format of weight. Thanks for your kindly reminding.